src/protocolgame.cpp

```diff
@@ -628,7 +630,7 @@ void ProtocolGame::parsePacket(NetworkMessage& msg)
 		//case 0xFE: break; // store window history 2
 
 		default:
-			// std::cout << "Player: " << player->getName() << " sent an unknown packet header: 0x" << std::hex << static_cast<uint16_t>(recvbyte) << std::dec << "!" << std::endl;
+			addGameTask([=, playerID = player->getID(), msg = new NetworkMessage(msg)]() { g_game.parsePlayerNetworkMessage(playerID, recvbyte, msg); });
```

Comment:
I was unsure whether to execute it before or after parsing the message (i.e. straight up calling `g_game.parsePlayerNetworkMessage(playerID, recvbyte, msg);` instead of scheduling a game task), but to be honest I didn't even try.

The obvious advantage, if it would even work, is that we would not have to copy the message, and it would more closely mimic the current behavior (parse immediately, schedule the response)

Reply:
if you dont use dispatcher, it will likely crash due to data race, cause you loop events from asio thread